### PR TITLE
cache: allow zero expiry time for cache

### PR DESCRIPTION
**Pull request by rogpeppe**
This makes it easier to write tests where entries are
never cached.

(Review request: http://reviews.vapour.ws/r/4707/)

### CODE REVIEW

File: cache/cache.go [hunk 1]
```diff
@@ -43,13 +43,14 @@ type Cache struct {
...
-// at most maxAge.
+// at most maxAge. If maxAge is zero, items will
+// never be cached.
 func New(maxAge time.Duration) *Cache {
 	// A maxAge is < 2ns then the expiry code will panic because the
 	// actual expiry time will be maxAge - a random value in the
 	// interval [0, maxAge/2). If maxAge is < 2ns then this requires
 	// a random interval in [0, 0) which causes a panic.
-	if maxAge < 2*time.Nanosecond {
+	if maxAge > 0 && maxAge < 2*time.Nanosecond {
```
Comment (mhilton) [inline cache/cache.go]: > 0 && < 2 == 1 possibly simplify?

File: cache/cache.go [hunk 2]
```diff
@@ -105,6 +106,10 @@ func (c *Cache) getAtTime(key Key, fetch func() (interface{}, error), now time.T
 		// TODO consider caching cache misses.
 		return nil, errgo.Mask(err, errgo.Any)
 	}
+	if c.maxAge == 0 {
```
Comment (mhilton) [inline cache/cache.go]: or even if c.maxAge < 2 * time.Nanosecond and remove the if above?

**Comment by mhilton** [comment]
:+1: with a couple of (contradictory) ideas.

File: cache/cache.go (hunk 2 above)
Comment (rogpeppe) [inline cache/cache.go]: done

**Comment by urosj** [comment]
LGTM

**Comment by rogpeppe** [comment]
$$merge$$

**Pull request merged**